### PR TITLE
Bring release/vsmac17.4 up to release/dev17.4

### PR DESCRIPTION
**Pull request by davidwengier**

### CODE REVIEW

**Review by sandyarmstrong — APPROVED** [review]

**Comment by jaredpar** [comment]
Is this change so big because it's including all the commits from razor-compiler that we merged in?

**Review by ryzngard — APPROVED** [review]

**Comment by davidwengier** [comment]
> Is this change so big because it's including all the commits from razor-compiler that we merged in?

This is Roslyn, not Razor 😛

Yesterday I created `release/vsmac17.4` for the purposes of servicing VS Mac, and based it on the commit of Roslyn that VS Mac 17.4 currently references. There is a desire however to not only service VS Mac with a few fixes, but also bring it up to level with VS Windows 17.4, so this PR is just pointing `release/vsmac17.4` to `release/dev17.4`. If I was a maintainer, I could have just done it with a push 😉

**Comment by sandyarmstrong** [comment]
> > Is this change so big because it's including all the commits from razor-compiler that we merged in?
> 
> This is Roslyn, not Razor 😛
> 
> Yesterday I created `release/vsmac17.4` for the purposes of servicing VS Mac, and based it on the commit of Roslyn that VS Mac 17.4 currently references. There is a desire however to not only service VS Mac with a few fixes, but also bring it up to level with VS Windows 17.4, so this PR is just pointing `release/vsmac17.4` to `release/dev17.4`. If I was a maintainer, I could have just done it with a push 😉

To give a little more context, VSMac 17.4 was previously referencing a more recent Roslyn, but experienced a perf regression, so rolled back to the last known "good" insertion prior to GA. The intent was always to get even again once we had a fix for the regression. From my testing, https://github.com/dotnet/roslyn/pull/65538 appears to be the fix we needed.

So once this merges, we'll be adding that revert, and also a couple of other fixes needed on the Mac side.

**Pull request merged**
